spec/models/miq_provision_virt_workflow_spec.rb

```diff
@@ -97,6 +97,13 @@
         expect(vlans.keys).to match_array(lan_keys)
         expect(vlans.values).to match_array(lan_keys)
       end
+
+      it '#load_hosts_vlans' do
+        hosts = [@host1]
+        MiqPreloader.preload(hosts, :lans => :switches)
+        expect(ActiveRecord::Base.connection).not_to receive(:exec_query)
+        workflow.load_hosts_vlans(hosts, {})
```

Comment:
This test doesn't make sense.  All the test is showing is that ActiveRecord works correctly.  A better test would be to use the query counting helper.

Reply:
See https://github.com/ManageIQ/manageiq/blob/master/spec/support/custom_matchers/exceed_query_limit.rb and https://github.com/ManageIQ/manageiq/blob/master/spec/support/custom_matchers/match_query_limit_of.rb custom matches for how to do query limit checks.